snippets/product-variant-options.liquid

```diff
@@ -45,10 +47,47 @@
     endfor
   -%}
 
-  {%- if block.settings.picker_type == 'button' -%}
+  {%- liquid
+    if value.swatch.image
+      assign image_url = value.swatch.image | image_url: width: 50
+      assign swatch_value = 'url(' | append: image_url | append: ')'
+    elsif value.swatch.color
+      assign swatch_value = 'rgb(' | append: value.swatch.color.rgb | append: ')'
+    else
+      assign swatch_value = nil
+    endif
+  -%}
+
+  {%- capture input_id -%}
+    {{ section.id }}-{{ option.position }}-{{ forloop.index0 }}
```

Comment:
Moving it to assign above makes it a really verbose declaration (because we need to use `append` to build the output). It's a good point about the newline that gets appended, fixing it using rstrip. (i.e `...forloop.index0 -}}`